types/mochawesome/index.d.ts

```diff
@@ -0,0 +1,6 @@
+// Type definitions for mochawesome/addContext 6.2
+// Project: https://github.com/adamgruber/mochawesome#readme
+// Definitions by: Chris Gilardi <https://github.com/Christop406>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare const unused: never;
```

Comment:
There actually is a module file at the `main` of the package: https://unpkg.com/browse/mochawesome@6.2.2/src/mochawesome.js

If none of this is public API, just replace this with `export {};`.

Reply:
> Further, you mention somewhere else in your review that I should name a file `addContext.d.ts`, would it be the same here, where I'll want to name the file `mochawesome.d.ts` to match the structure of the original project?

Technically that would be most correct, except that our publishing infrastructure is set up to assume and enforce that the main entrypoint is named `index.d.ts`. So you could name the main file `mochawesome.d.ts` and then re-export that module from `index.d.ts` if you want, but I don’t think it’s necessary, since users should really only reference the main entrypoint by `"mochawesome"`. Also, note that what is exported from this main entrypoint is never mentioned in the project’s README, so my assumption was it’s an internal API designed to be consumed by mocha itself, not users. So I don’t think it’s necessary to put anything except `export {}` here. I only mentioned it because `declare const unused: never` isn’t a good substitute for “nothing” since it introduces a global named `unused`, and you have to have an `index.d.ts` file just because the publishing infrastructure says so.

> And further, is it worth making modules for the rest of the files in that directory as well, even if they are not meant to be public API?

Nope, you can/should ignore those.

Reply:
@andrewbranch thanks again for your thorough responses. I actually did end up writing the typings for the `Mochawesome` constructor function, because it is the best match to how the package works in JS land.

To test, I made a test node package and tried importing from `mochawesome`, and was able to create a new `Mochawesome` instance using the import. In the interest of keeping the public API as close to what is actually available as possible, I think it makes sense to include that `Mochawesome` class as an export from this types package. I feel that there are possible uses where users may want to access the raw `Mochawesome` instances, and adding these types in supports it.

The test looked somewhat like this (abbreviated):

```js
const Mocha = require('mocha');
const mochawesome = require('mochawesome');
const suite = new Mocha.Suite('Test Suite');

// ... suite setup

const runner = new Mocha.Runner(suite, false)
runner.run();

const m = new mochawesome(runner, {});

runner.on('end', () => {
    // The following properties are populated and accessible via IntelliSense (the public API of the class)
    console.log(m.output);
    console.log(m.done);
    console.log(m.config);
});
```

So I think it makes sense to include it here to at least have feature parity with the JS equivalent.

If you think I'm in the wrong here though, that's fine too.